precompiles/regular/regular.go

```diff
@@ -0,0 +1,248 @@
+package regular
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/vm"
+
+	ptypes "github.com/zeta-chain/zetacore/precompiles/types"
+	"github.com/zeta-chain/zetacore/testutil/contracts"
+	fungiblekeeper "github.com/zeta-chain/zetacore/x/fungible/keeper"
+	fungibletypes "github.com/zeta-chain/zetacore/x/fungible/types"
+)
+
+const (
+	RegularCallMethodName     = "regularCall"
+	Bech32ToHexAddrMethodName = "bech32ToHexAddr"
+	Bech32ifyMethodName       = "bech32ify"
+)
+
+var (
+	ABI                 abi.ABI
+	ContractAddress     = common.BytesToAddress([]byte{101})
+	GasRequiredByMethod = map[[4]byte]uint64{}
+	ExampleABI          *abi.ABI
+)
+
+func init() {
+	ABI, GasRequiredByMethod = initABI()
+	ExampleABI, _ = contracts.ExampleMetaData.GetAbi()
+}
+
+var RegularModuleMetaData = &bind.MetaData{
+	ABI: "[{\"inputs\":[{\"internalType\":\"string\",\"name\":\"bech32\",\"type\":\"string\"}],\"name\":\"bech32ToHexAddr\",\"outputs\":[{\"internalType\":\"address\",\"name\":\"addr\",\"type\":\"address\"}],\"stateMutability\":\"view\",\"type\":\"function\"},{\"inputs\":[{\"internalType\":\"string\",\"name\":\"prefix\",\"type\":\"string\"},{\"internalType\":\"address\",\"name\":\"addr\",\"type\":\"address\"}],\"name\":\"bech32ify\",\"outputs\":[{\"internalType\":\"string\",\"name\":\"bech32\",\"type\":\"string\"}],\"stateMutability\":\"view\",\"type\":\"function\"},{\"inputs\":[{\"internalType\":\"string\",\"name\":\"method\",\"type\":\"string\"},{\"internalType\":\"address\",\"name\":\"addr\",\"type\":\"address\"}],\"name\":\"regularCall\",\"outputs\":[{\"internalType\":\"uint256\",\"name\":\"result\",\"type\":\"uint256\"}],\"stateMutability\":\"nonpayable\",\"type\":\"function\"}]",
+}
+
+func initABI() (abi abi.ABI, gasRequiredByMethod map[[4]byte]uint64) {
+	gasRequiredByMethod = map[[4]byte]uint64{}
+	if err := abi.UnmarshalJSON([]byte(RegularModuleMetaData.ABI)); err != nil {
+		panic(err)
+	}
+	for methodName := range abi.Methods {
+		var methodID [4]byte
+		copy(methodID[:], abi.Methods[methodName].ID[:4])
+		switch methodName {
+		case RegularCallMethodName:
+			gasRequiredByMethod[methodID] = 10
```

Comment:
can you explain `10` here? Also consider moving this to const

Reply:
It's just the gas used in this specific method. Godo idea to move it to `RegularCallMethodGasRequired`.